support/metrics/metricsTracker.go

```diff
@@ -147,6 +148,10 @@ func (mt *MetricsTracker) SendDeleteEvent(exit bool) error {
 }
 
 func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) error {
+	if mt.apiKey == "" {
+		return fmt.Errorf("could not send amplitude event of type %s due to undefined API key", eventType)
```

Comment:
@debnil this should return `nil` not an error.

this is the scenario where we build without the api key, i.e. by compiling from source. If you run this bot now by compiling from source, I'd imagine that the bot is unable to start up because of this error?